File: app/src/main/java/com/yvrun/officeprocess/mvp/model/BaseModel.java
package com.yvrun.officeprocess.mvp.model;

import com.yvrun.officeprocess.mvp.net.RetrofitService;

import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;

public abstract class BaseModel<T>{

    protected RetrofitService mRetrofitService;

    public BaseModel(RetrofitService retrofitService) {
        mRetrofitService = retrofitService;
    }

    public Observable<T> getData(){
        return null;
    }
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/mvp/model/BaseModel.kt
package com.chengzhen.kotlinappother.mvp.model


import com.chengzhen.kotlinappother.mvp.net.RetrofitService

import io.reactivex.Observable
import io.reactivex.ObservableEmitter
import io.reactivex.ObservableOnSubscribe

abstract class BaseModel<T>(protected var mRetrofitService: RetrofitService) {

    val data: Observable<T>?
        get() = null
}
